olive/engine/footprint.py

```diff
@@ -0,0 +1,178 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from collections import OrderedDict
+from datetime import datetime
+from pathlib import Path
+
+from olive.common.config_utils import ConfigBase, config_json_dumps, config_json_loads
+
+logger = logging.getLogger(__name__)
+
+
+class FootprintNodeMetric(ConfigBase):
+    value: dict
+    cmp_direction: dict = {}
+    is_goals_met: bool = False
+
+
+class FootprintNode(ConfigBase):
+    # None for no parent which means current model is the input model
+    parent_model_id: str = None
+    model_id: str
+    model_config: dict = None
+    from_pass: str = None
+    pass_run_config: dict = None
+    is_pareto_frontier: bool = False
+    metrics: FootprintNodeMetric = FootprintNodeMetric(value={}, is_goals_met=False)
+
+    date_time: float = datetime.now().timestamp()
+
+    def update(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+
+class Footprint:
```

Comment:
Can you add some unit tests for this class?

Reply:
Sure, will add it. Also for the documents for olive footprints, will update it later when all comments are solved.